OrbitCore/OrbitFunction.h

```diff
@@ -132,29 +81,103 @@ class Function {
 
- public:  // TODO...
-  std::string m_Name;
-  std::string m_PrettyName;
-  std::string m_PrettyNameLower;
-  std::string m_Module;
-  std::string m_File;
-  std::string m_Probe;
-  uint64_t m_Address = 0;
-  uint64_t m_ModBase = 0;
-  uint32_t m_Size = 0;
-  uint32_t m_Id = 0;
-  uint32_t m_ParentId = 0;
-  int m_Line = 0;
-  int m_CallConv = -1;
-  std::vector<FunctionParam> m_Params;
-  std::vector<Argument> m_ArgInfo;
-  Pdb* m_Pdb = nullptr;
-  uint64_t m_NameHash = 0;
-  OrbitType m_OrbitType = NONE;
-  std::shared_ptr<FunctionStats> m_Stats;
-
- protected:
-  bool m_Selected = false;
+  void Print();
+  void SetAsMainFrameFunction();
+  void AddParameter(const FunctionParam& param) {
+    params_.push_back(param);
+  }
+
+  // TODO: It looks like most setters are used by TestRemoteMessages::Run()
+  // only. Move these to a constructor?
+  void SetName(const std::string& name) { name_ = name; }
+  void SetPrettyName(const std::string& pretty_name) {
+    pretty_name_ = pretty_name;
+  }
+  void SetAddress(uint64_t address) { address_ = address; }
+  void SetModule(const std::string& module) { module_ = module; }
+  void SetFile(const std::string& file) { file_ = file; }
+  void SetProbe(const std::string& probe) { probe_ = probe; }
+  void SetId(uint32_t id) { id_ = id; }
+  void SetParentId(uint32_t parent_id) { parent_id_ = parent_id; }
+  void SetSize(uint64_t size) { size_ = size; }
+  void SetLine(uint32_t line) { line_ = line; }
+  void SetCallingConvention(int calling_convention) {
+    calling_convention_ = calling_convention;
+  }
+  void SetOrbitType(OrbitType type) { type_ = type; }
+  void SetPdb(Pdb* pdb) { pdb_ = pdb; }
+
+  const std::string& Name() const { return name_; }
+  const std::string& PrettyName() const;
+  const std::string& Lower() {
+    if (pretty_name_lower_.size() == 0) {
+      pretty_name_lower_ = ToLower(pretty_name_);
+    }
+    return pretty_name_lower_;
+  }
+  OrbitType GetOrbitType() const { return type_; }
+  uint32_t Line() const { return line_; }
+  uint64_t Size() const { return size_; }
+  const std::string& Probe() const { return probe_; }
+  int CallingConvention() const { return calling_convention_; }
+  const Pdb* GetPdb() const { return pdb_; }
+  const FunctionStats* Stats() const { return stats_.get(); }
+  const char* GetCallingConventionString();
+  void ProcessArgumentInfo();
+  bool IsMemberFunction();
+  uint64_t Hash() const { return StringHash(pretty_name_); }
+  void UpdateStats(const Timer& timer);
+  bool Hookable();
+  void Select();
+  void PreHook();
+  void UnSelect();
+  void ToggleSelect() { /*if( Hookable() )*/
+    selected_ = !selected_;
+  }
+  bool IsSelected() const { return selected_; }
+  // Calculates and returns the absolute address of the function.
+  uint64_t GetVirtualAddress() const;
+  uint64_t Address() const { return address_; }
+  const std::string& File() const { return file_; }
+  bool IsOrbitFunc() { return type_ != OrbitType::NONE; }
```

Comment:
Maybe we should make all the Is<something>() functions const?

Reply:
done